AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Models/AWSCognitoUserPoolTokens.swift

```diff
@@ -38,6 +38,32 @@ public struct AWSCognitoUserPoolTokens: AuthCognitoTokens {
         self.expiration = expiration
     }
 
+    public init(idToken: String,
+                accessToken: String,
+                refreshToken: String,
+                expiresIn: Int? = nil) {
+
+        if let expiresIn =  expiresIn {
+            self.init(idToken: idToken,
+                      accessToken: accessToken,
+                      refreshToken: refreshToken,
+                      expiresIn: expiresIn)
+        } else {
+            // If for some reason, failed to extract "exp" value, use the value as zero and don't block the user
+            var expirationDoubleValue: Double = 0
```

Comment:
if we can't extract the "exp" values, how does expiresIn in 0 get used later on? what's the impact of setting it to 0?

Reply:
There is no impact. As we are not expecting customers to use it.. 
So I have also marked this `expiry` property as deprecated.

Reply:
i meant, the expiration property is set to 0, and then how does the plugin use this value later on in the implementation? i can see that customers wouldn't need to read the value or do anything with it

Reply:
Even the plugin doesn't use it in any way.. Its just something Cognito has and we passed it back to the customer.. Hoping to remove it in the next major version.